recipes/acado/all/conanfile.py

```diff
@@ -0,0 +1,113 @@
+import os
+import glob
+
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class AcadoConan(ConanFile):
+    name = "acado"
+    description = "ACADO Toolkit is a software environment and algorithm collection for automatic control and dynamic optimization."
+    license = "LGPL-3.0"
+    topics = ("conan", "acado", "control", "optimization", "mpc")
+    homepage = "https://github.com/acado/acado"
+    url = "https://github.com/conan-io/conan-center-index"
+    exports_sources = ["CMakeLists.txt", "cmake/qpoases.cmake", "patches/**"]
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "codegen_only": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "codegen_only": True,
+    }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = glob.glob("acado-*/")[0]
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+
+        self._cmake.definitions["ACADO_BUILD_SHARED"] = self.options.shared
+        self._cmake.definitions["ACADO_BUILD_STATIC"] = not self.options.shared
+
+        self._cmake.definitions["ACADO_WITH_EXAMPLES"] = False
+        self._cmake.definitions["ACADO_WITH_TESTING"] = False
+        self._cmake.definitions["ACADO_DEVELOPER"] = False
+        self._cmake.definitions["ACADO_INTERNAL"] = False
+        self._cmake.definitions["ACADO_BUILD_CGT_ONLY"] = self.options.codegen_only
+
+        self._cmake.configure()
```

Comment:
```suggestion
        self._cmake.configure(build_folder=self._build_subfolder)
```

Reply:
[The docu](https://docs.conan.io/en/latest/reference/build_helpers/cmake.html#cmake) mentions this:
```python
        # same as cmake.configure(source_folder=self.source_folder, build_folder=self.build_folder)
        cmake.configure()
```

Reply:
`self.build_folder` != `self._build_subfolder`

build_folder is a cache build folder, _build_subfolder is a subfolder there, used to separate binary cache from source files. More or less when you create a build only to store all artifacts e.g. 

    mkdir build && cd build
    cmake ..
    cmake --build .
    
The folder build/ contains all binary stuff, but the source folder is preserved.